raiden/network/transport/matrix.py

```diff
@@ -1349,52 +1387,55 @@ def _get_room_ids_for_address(
         If filter_private=None, filter according to self._private_rooms
         """
         address_hex: AddressHex = to_checksum_address(address)
-        room_ids = self._client.account_data.get(
-            'network.raiden.rooms',
-            {},
-        ).get(address_hex)
-        if not room_ids:  # None or empty
-            room_ids = list()
-        if not isinstance(room_ids, list):  # old version, single room
-            room_ids = [room_ids]
-
-        if filter_private is None:
-            filter_private = self._private_rooms
-        if not filter_private:
-            # existing rooms
-            room_ids = [
-                room_id
-                for room_id in room_ids
-                if room_id in self._client.rooms
-            ]
-        else:
-            # existing and private rooms
-            room_ids = [
-                room_id
-                for room_id in room_ids
-                if room_id in self._client.rooms and self._client.rooms[room_id].invite_only
-            ]
+        with self._account_data_lock:
+            room_ids = self._client.account_data.get(
+                'network.raiden.rooms',
+                {},
+            ).get(address_hex)
+            self.log.debug('acct data get', room_ids=room_ids, for_address=address_hex)
+            if not room_ids:  # None or empty
+                room_ids = list()
+            if not isinstance(room_ids, list):  # old version, single room
+                room_ids = [room_ids]
+
+            if filter_private is None:
+                filter_private = self._private_rooms
+            if not filter_private:
+                # existing rooms
+                room_ids = [
+                    room_id
+                    for room_id in room_ids
+                    if room_id in self._client.rooms
+                ]
+            else:
+                # existing and private rooms
+                room_ids = [
+                    room_id
+                    for room_id in room_ids
+                    if room_id in self._client.rooms and self._client.rooms[room_id].invite_only
+                ]
+
+            return room_ids
+
+    def _leave_unused_rooms(self, _address_to_room_ids: Dict[AddressHex, List[_RoomID]]):
+        """
+        Checks for rooms we've joined and which partner isn't health-checked and leave.
+
+        **MUST** be called from a context that holds the `_account_data_lock`.
+        """
+        _msg = '_leave_unused_rooms called without account data lock'
+        assert self._account_data_lock.locked(), _msg
 
-        return room_ids
+        # TODO: Remove the next two lines and check if transfers start hanging again
+        self._client.set_account_data('network.raiden.rooms', _address_to_room_ids)
+        return
```

Comment:
From what I see, you effectivelly disabled `_leave_unused_rooms` here. Could you please just keep the logic as it was, and instead replace the call to it with simpler `_client.set_account_data`? it'll be clearer what it's doing

Reply:
Yes that effectively disables leaving rooms. I don’t think this is a problem for red eyes. The whole feature is only relevant to protect against (D)DoS attacks.

Reply:
I see. If that is so can you clear this up with a comment there and make an issue tagged with Ithaca so that we are all aware of it?

Reply:
I created the issue https://github.com/raiden-network/raiden/issues/3262